### PR TITLE
document outbound route metrics

### PR DESCRIPTION
**Pull request by hawkw**
This branch adds documentation for the new metrics added in linkerd/linkerd2-proxy#2377 and linkerd/linkerd2-proxy#2380.

### CODE REVIEW

File: linkerd.io/content/2-edge/reference/proxy-metrics.md [hunk 1]
```diff
@@ -229,14 +229,54 @@ connection closes (`tcp_close_total` and `tcp_connection_duration_ms`):
...
+
+### Labels
+
+Each of these metrics has the following common labels, which describe the
+Kubernetes resources to which traffic is routed by the proxy:
+
+* `parent_group`, `parent_kind`, `parent_name`, and `parent_namespace` reference
+  the parent resource through which the proxy discovered the route binding.
+  The parent resource of an HTTPRoute is generally a Service.
+* `route_grouo`, `route_kind`, `route_name`, and `route_namespace` reference the
```
Comment (adleong) [inline linkerd.io/content/2-edge/reference/proxy-metrics.md:262]: typo: route_grouo

File: linkerd.io/content/2-edge/reference/proxy-metrics.md [hunk 2]
```diff
@@ -229,14 +229,54 @@ connection closes (`tcp_close_total` and `tcp_connection_duration_ms`):
...
+  route resource through which the proxy discovered the route binding. This will
+  either reference an HTTPRoute resource or a default (synthesized) route.
+* `backend_grouo`, `backend_kind`, `backend_name`, and `backend_namespace`
```
Comment (adleong) [inline linkerd.io/content/2-edge/reference/proxy-metrics.md:265]: ditto

File: linkerd.io/content/2-edge/reference/proxy-metrics.md [hunk 3]
```diff
@@ -229,14 +229,54 @@ connection closes (`tcp_close_total` and `tcp_connection_duration_ms`):
...
 * `identity_cert_expiration_timestamp_seconds`: A gauge of the time when the
   proxy's current mTLS identity certificate will expire (in seconds since the UNIX
   epoch).
 * `identity_cert_refresh_count`: A counter of the total number of times the
   proxy's mTLS identity certificate has been refreshed by the Identity service.
+
+## Outbound `xRoute` Metrics
+
+When performing policy-based routing, proxies may dispatch requests through
+per-route backend configurations. In order to record how routing rules
```
Comment (hawkw) [inline linkerd.io/content/2-edge/reference/proxy-metrics.md:243]: would be nice if this linked to the HTTPRoute docs once they land

File: linkerd.io/content/2-edge/reference/proxy-metrics.md [hunk 4]
```diff
@@ -229,14 +229,54 @@ connection closes (`tcp_close_total` and `tcp_connection_duration_ms`):
...
+In addition, the `outbound_http_balancer_endpoints` gauge metric adds the
+following labels:
+
+* `endpoint_state`: Either "ready" if the endpoint is available to have requests
+  routed to it by the load balancer, or "pending" if the endpoint is currently
+  unavailable.
+
+  Endpoints may be "pending" when a connection is being established (or
+  reestablished), or when the endpoint has been made unavailable by failure
+  accrual.
```
Comment (hawkw) [inline linkerd.io/content/2-edge/reference/proxy-metrics.md:278]: would be nice if this linked to the failure accrual docs once #1598 lands

**Review by adleong — APPROVED** [review]

**Pull request merged**